contracts/AllPayAuction.sol

```diff
@@ -1,193 +1,179 @@
     }
 
-    uint256 private auctionCounter;
-    mapping(uint256 => Auction) public auctions;
-
     event AuctionCreated(
-        uint256 indexed auctionId,
+        uint256 indexed Id,
         string name,
         string description,
-        string imageUrl,
+        string imgUrl,
+        address auctioneer,
         AuctionType auctionType,
-        address indexed auctioneer,
-        address auctionedTokenAddress,
+        address auctionedToken,
         uint256 auctionedTokenIdOrAmount,
+        address biddingToken,
         uint256 startingBid,
-        uint256 highestBid,
-        uint256 deadline
-    );
-    event BidPlaced(
-        uint256 indexed auctionId,
-        address indexed bidder,
-        uint256 bidAmount
-    );
-    event ItemWithdrawn(
-        uint256 indexed auctionId,
-        address indexed winner,
-        uint256 winningBid
-    );
-    event FundsWithdrawn(
-        uint256 indexed auctionId,
-        address indexed auctioneer,
-        uint256 amount
+        uint256 minBidDelta,
+        uint256 deadline,
+        uint256 deadlineExtension
     );
 
-    modifier onlyActiveAuction(uint256 auctionId) {
-        require(
-            block.timestamp < auctions[auctionId].deadline,
-            "Auction has expired"
-        );
+    modifier validAuctionId(uint256 auctionId) {
+        require(auctionId>=0 && auctionId < auctionCounter,"Invalid auctionId");
         _;
     }
 
     function createAuction(
         string memory name,
         string memory description,
-        string memory imageUrl,
+        string memory imgUrl,
         AuctionType auctionType,
-        address auctionedTokenAddress,
+        address auctionedToken,
         uint256 auctionedTokenIdOrAmount,
+        address biddingToken,
         uint256 startingBid,
         uint256 minBidDelta,
-        uint256 deadlineExtension,
-        uint256 duration
-    ) external {
-        require(bytes(name).length > 0, "Name cannot be empty");
-        require(startingBid > 0, "Starting bid must be greater than 0");
-        require(minBidDelta > 0, "Minimum bid delta must be greater than 0");
-
-        if (auctionType == AuctionType.NFT) {
-            require(IERC721(auctionedTokenAddress).ownerOf(auctionedTokenIdOrAmount) == msg.sender,"Caller must own the NFT");
-            IERC721(auctionedTokenAddress).transferFrom(msg.sender,address(this),auctionedTokenIdOrAmount);
-        } else {
-            require(IERC20(auctionedTokenAddress).balanceOf(msg.sender) >= auctionedTokenIdOrAmount,"Caller must have sufficient tokens");
-            IERC20(auctionedTokenAddress).transferFrom(msg.sender,address(this),auctionedTokenIdOrAmount);
+        uint256 duration,
+        uint256 deadlineExtension
+    ) external{
+        require(bytes(name).length>0,"Name must be present");
+        require(minBidDelta>=0,"Minimum bid delta cannot be negative");
+        require(startingBid>=0,"Starting bid cannot be negative");
```

Comment:
These "requires" are not necessary. Since `minBidDelta` and `startingBid` are uint256, they cannot be negative anyway.